### DISCUSSION TITLE
why many scheduled task run immidiately when resume after pause?

### DISCUSSION THREAD

**Discussion by virtouso** [quartznet/quartznet#1706] [General]
i need the exact pause/resume mechanics from quartz. when its paused, i need to continue from the time scheduler paused but many scheduled tasks that their time has passed will be called.

for example when i call `     await scheduler.ResumeAll();`   i see this

 moeen task ran8|17
 moeen task ran9|17
 moeen task ran10|17
 moeen task ran11|17
 moeen task ran13|17
 moeen task ran12|17
 moeen task ran15|17
 moeen task ran14|17
 moeen task ran16|17
 moeen task ran17|17
 moeen task ran18|18
 moeen task ran19|19
 moeen task ran20|20
 moeen task ran21|21

all passed tasks get called at same time but i need them to be called from the time i paused/ how can i do that?

**Comment by lahma** [comment]
I think you are experiencing misfire behavior. See documentation about the options what to do when Quartz detects a missed fire (in your case it tries to catch up and run the triggers as fast as it can).

**Reply by virtouso** [reply]
thanks for the replay but I don't want to ignore miss fires. I just want all remaining jobs run from the time i paused. for example when pausing one job had to run one second after a pause. now I need that one to run one second after resume and any other one from the time i resumed.

**Reply by lahma** [reply]
I don't think that's quite possible. Trigger was meant to run at the next second just after the pause, resume cannot coordinate such schedule as it cannot be calculated from start time + expected intervals.

**Reply by virtouso** [reply]
i juse figured it out. i needed rescheduling:



      ```
      var allTriggerKeys = await scheduler.GetTriggerKeys(GroupMatcher<TriggerKey>.AnyGroup());

            foreach (var triggerKey in allTriggerKeys)
            {
                var triggerdetails = scheduler.GetTrigger(triggerKey);
                var Jobdetails = scheduler.GetJobDetail(triggerdetails.Result.JobKey);


                var time =  triggerdetails.Result.StartTimeUtc.DateTime.AddSeconds((DateTime.Now-pauseTime).Seconds);

                ITrigger trigger = TriggerBuilder.Create()
        .WithIdentity(triggerKey.Name, "group").StartAt(time).Build();

                await scheduler.RescheduleJob(triggerKey, trigger);

            }

            await scheduler.ResumeAll();
```

**Reply by lahma** [reply]
I didn't suggest reschedule because that would be a lot handling to do just for each pause and resume if there are a lot of triggers, but good if that works for you. Also try to avoid the use of `.Result`, always `await` to get the result or if really necessary, use `.GetAwaiter().GetResult()`.
